src/ProductTour/index.jsx

```diff
@@ -82,15 +82,17 @@ const ProductTour = React.forwardRef(({ tours }, ref) => {
     setCurrentCheckpointData(null);
   };
 
-  const handleEnd = () => {
+  const handleEnd = (checkpointIndex) => {
     setIndex(0);
     setIsTourEnabled(false);
     if (onEnd) {
-      onEnd();
+      onEnd(prunedCheckpoints[checkpointIndex]);
+    }
+    if (prunedCheckpoints[checkpointIndex].onEnd) {
```

Comment:
[question] Is `onEnd` in the prop types for the `checkpoints` array to define that it should be a function?